### PR TITLE
build(deps): bump Vert.x from 4.1.0 to 4.2.1 

### PR DESCRIPTION
**Pull request by pk-work**

### CODE REVIEW

File: src/test/java/io/neonbee/internal/ClassTemplate.java [hunk 1]
```diff
@@ -79,12 +92,30 @@ private String resourcePath() {
...
+
+        JavaFileObject javaFileToCompile = standardFileManager.getJavaFileForInput(SOURCE_PATH, getClassName(), SOURCE);
+
+        DiagnosticCollector<JavaFileObject> diagnostics = new DiagnosticCollector<>();
+        JavaCompiler.CompilationTask task =
+                javaCompiler.getTask(null, standardFileManager, diagnostics, null, null, List.of(javaFileToCompile));
+
+        if (!task.call()) {
+            Diagnostic<?> d = diagnostics.getDiagnostics().get(0);
+            throw new RuntimeException("Compilation Failed: \n" + d.getMessage(Locale.getDefault()));
```
Comment (s4heid) [inline src/test/java/io/neonbee/internal/ClassTemplate.java:115]:
```suggestion
            String messages = diagnostics.getDiagnostics().stream().map(d -> d.getMessage(Locale.getDefault())).collect(joining("\n"));
            throw new RuntimeException("Compilation Failed: \n" + messages);
```

**Review by s4heid — CHANGES_REQUESTED** [review]

File: src/test/java/io/neonbee/internal/ClassTemplate.java [hunk 2]
```diff
@@ -79,12 +92,30 @@ private String resourcePath() {
...
+
         Files.writeString(sourceFile, reifyTemplate());
 
-        try (URLClassLoader urlc =
-                new URLClassLoader(new URL[] { tempDir.toUri().toURL() }, ClassLoader.getSystemClassLoader())) {
-            CompilingClassLoader compilingLoader = new CompilingClassLoader(urlc, resourcePath());
-            return compilingLoader.getClassBytes(getClassName());
+        JavaCompiler javaCompiler = ToolProvider.getSystemJavaCompiler();
+        if (javaCompiler == null) {
+            throw new RuntimeException("Unable to detect java compiler, make sure you're using a JDK not a JRE!");
```
Comment (kristian) [inline src/test/java/io/neonbee/internal/ClassTemplate.java:100]: "Java" not "java"

**Review by kristian — APPROVED** [review]
lgtm

File: src/test/java/io/neonbee/internal/ClassTemplate.java (hunk 1 above)
Comment (pk-work) [reply src/test/java/io/neonbee/internal/ClassTemplate.java]: done

File: src/test/java/io/neonbee/internal/ClassTemplate.java (hunk 2 above)
Comment (pk-work) [reply src/test/java/io/neonbee/internal/ClassTemplate.java]: done

**Review by s4heid — APPROVED** [review]
lgtm

**Pull request merged**